File: framework/widget/src/main/java/com/sofar/widget/recycler/adapter/multitype/Type.java
package com.sofar.widget.recycler.adapter.multitype;

import com.sofar.widget.recycler.adapter.Cell;

public class Type {

  public final Class mClass;
  public final Cell mCell;
  public final Linker mLinker;

  public Type(Class clazz, Cell cell, Linker linker) {
    mClass = clazz;
    mCell = cell;
    mLinker = linker;
  }
}

File: core/ui/src/main/java/com/sofar/core/ui/recyclerview/adapter/multitype/Type.kt
package com.sofar.core.ui.recyclerview.adapter.multitype

/**
 * 存储数据模型、单元格工厂以及路由选择器的映射关系实体
 */
data class Type(
  val mClass: Class<*>,
  val mFactory: CellFactory<*>,
  val mLinker: Linker<*>
)
